capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -326,25 +331,31 @@ export class SearchUserModal extends Component {
                                 {user.email}
                               </Card.Text>
 
-
-                              {this.arrayContainsJSONId(this.state.addedUsers, user) ?
-                                <Button className='my-2 w-75' variant='danger' onClick={() => this.removeUserFromAddedUsers(user)}>
-                                  {this.props.removeBtnText || 'Remove'}
-                                </Button>
-                                :
-                                <Button className='my-2 w-75' onClick={() => this.addUserToAddedUsers(user)}>
-                                  {this.props.addBtnText || 'Add'}
-                                </Button>
-                              }
-                              <br />
-                              {this.arrayContainsJSONId(this.state.addedFriends, user) ?
-                                <Button className='mt-2 mb-1 w-75' variant='danger' onClick={() => this.removeUserFromAddedFriends(user)}>
-                                  Remove Friend
+                              {this.props.userType !== 'viewer' &&
+                                <>
+                                  {this.arrayContainsJSONId(this.state.addedUsers, user) ?
+                                    <Button className='my-2 w-75' variant='danger' onClick={() => this.removeUserFromAddedUsers(user)}>
+                                      {this.props.removeBtnText || 'Remove'}
+                                    </Button>
+                                    :
+                                    <Button className='my-2 w-75' onClick={() => this.addUserToAddedUsers(user)}>
+                                      {this.props.addBtnText || 'Add'}
+                                    </Button>}
+                                  <br />
+                                </>}
+
+                              {user.id !== window.localStorage.getItem('userID') &&
```

Comment:
Nice check to prevent showing the friend buttons if this is the user's own card!